### PR TITLE
Fix/12245

### PR DESCRIPTION
**Pull request by Mr-Sunglasses**
fix: #12245 
- Added `default=Exception` to the `_BaseExceptionT` type variable in `ExceptionGroup`.

### CODE REVIEW

**Comment by Mr-Sunglasses** [comment]
Some tests fail due to #12255 which will be fixed in #12257

**Review by srittau — APPROVED** [review]
Thanks!

File: stdlib/builtins.pyi [hunk 1]
```diff
@@ -2007,7 +2007,7 @@ if sys.version_info >= (3, 10):
 
 if sys.version_info >= (3, 11):
     _BaseExceptionT_co = TypeVar("_BaseExceptionT_co", bound=BaseException, covariant=True)
-    _BaseExceptionT = TypeVar("_BaseExceptionT", bound=BaseException)
+    _BaseExceptionT = TypeVar("_BaseExceptionT", bound=BaseException, default=Exception)
```
[the remark this answers is not in the document]
Comment (Mr-Sunglasses) [reply stdlib/builtins.pyi:2010]: Yeah, it seems the defaults should indeed be applied to `_BaseExceptionT_co` and `_ExceptionT_co` , I've added to them.

File: stdlib/builtins.pyi [hunk 2]
```diff
@@ -2006,9 +2006,9 @@ if sys.version_info >= (3, 10):
     class EncodingWarning(Warning): ...
 
 if sys.version_info >= (3, 11):
-    _BaseExceptionT_co = TypeVar("_BaseExceptionT_co", bound=BaseException, covariant=True)
+    _BaseExceptionT_co = TypeVar("_BaseExceptionT_co", bound=BaseException, covariant=True, default=Exception)
```
Comment (JelleZijlstra) [inline stdlib/builtins.pyi:2009]: This should be BaseException.

Also, this seems tricky enough that I think it's worth adding some tests.

**Pull request merged**